core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -400,7 +400,14 @@ public Transaction newTransaction() {
 
   @Override
   public StaticTableOperations operations() {
-    return (StaticTableOperations) ((BaseTable) lazyTable()).operations();
+    if (lazyTable() instanceof BaseTable) {
+      BaseTable table = (BaseTable) lazyTable();
+      return (StaticTableOperations) table.operations();
+    } else if (lazyTable() instanceof BaseMetadataTable) {
+      BaseMetadataTable table = (BaseMetadataTable) lazyTable();
+      return (StaticTableOperations) table.table().operations();
+    }
+    return null;
```

Comment:
In what case, `null` would be returned. it might need to NPE in other places.. would having strict check preferred in this case?

Reply:
actually we shouldn't be running into the `null` case. I've updated the code to reflect that